File: 3370-smallest-number-with-all-set-bits/3370-smallest-number-with-all-set-bits.js
/**
 * @param {number} n
 * @return {number}
 */
var smallestNumber = function(n) {
    let binary = "";

    while(n > 0){
        binary = (n%2) + binary;
        n = Math.floor(n/2);
    }

    binary = Array(binary.length).fill('1').join("");
    let decimal = parseInt(binary,2);
    return decimal;
};

File: 3370-smallest-number-with-all-set-bits/3370-smallest-number-with-all-set-bits.cpp
class Solution {
public:
    int smallestNumber(int n) {
        string binary = "";

        while(n > 0){
            binary = to_string(n%2) + binary;
            n /= 2;
        }

        for(int i = 0; i < binary.size(); i++){
            binary[i] = '1';
        }

        int decimal = stoi(binary,0,2);

        return decimal;
    }
};